File: lc003.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size();
        bool charExist[256];
        memset(charExist, false, sizeof(charExist));
        int res = 0;
        int start = 0;
        int end = 0;
        while(start < n && end < n){
            if(charExist[s[end]] == false){
                charExist[s[end]] = true;
                end++;
            }
            else{
                charExist[s[start]] = false;
                start++;
            }
            res = max(res, end - start);
        }
        return res;
    }
};

File: lc003.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        start = 0
        end = 0
        myDict = {}
        res = 0
        n = len(s)
        if(s == ""):
            return 0
        while(end < n):
            if(myDict.get(s[end]) != None):
                del myDict[s[start]]
                start = start + 1
            else:
                myDict[s[end]] = 1
                end = end + 1
                res = max(res, (end - start))
        return res
            
